website/addons/dryad/provider.py

```diff
@@ -0,0 +1,337 @@
+from website.addons.dryad.settings import DRYAD_BASE_URL
+from website.citations.providers import CitationsOauthProvider
+from website.addons.dryad.serializer import DryadSerializer
+import requests
+from furl import furl
+import xml
+import re
+
+class DryadProvider(CitationsOauthProvider):
+    name = 'Dryad'
+    short_name = 'dryad'
+    serializer = DryadSerializer
+
+    def get(self, url, params={}):
+        if params:
+            r = requests.get(url, params=params)
+        else:
+            r = requests.get(url)
+        return r
+
+    def post(self, url):
+        r = requests.post(url)
+        return r
+
+    def get_dryad_metadata(self, doi):
+        """ Retrieves the metadata of a dryad item in the form of xml.Document
+        :param doi: Dryad DOI in the form of "doi:10.5061/dryad.XXXX"
+        :type doi: string
+        :returns:  xml.dom.minidom.Document -- xml form of the dryad metadata
+        :raises: urllib.error.HTTPError
+        """
+        dryad_metadata_path = furl(DRYAD_BASE_URL)
+        dryad_metadata_path.path.segments = ['mn', 'object', doi]
+        resp = self.get(dryad_metadata_path.url)
+        # The below encoding is mandatory
+        return xml.dom.minidom.parseString(resp.text.encode('utf-8'))
+
+    def get_dryad_page(self, doi='doi:10.5061/dryad.1850'):
+        """Retrieves the html of the dryad page for a package. As some of the
+        object metadata is served directly to the page, this kludge is used
+        to scrape that data off the page.
+        :param doi: Dryad DOI in the form of "doi:10.5061/dryad.XXXX"
+        :type doi: string
+        :returns:  beautifulsoup4.BeautifulSoup -- soup form of the dryad metadata
+        :raises: urllib.error.HTTPError
+        """
+        dryad_page_path = furl(DRYAD_BASE_URL)
+        dryad_page_path.path.segments = ['resource', doi]
+        resp = self.get(dryad_page_path.url)
+        # The below encoding is mandatory
+        return resp.text
+
+    def extract_dryad_title(self, metadata):
+        return metadata.getElementsByTagName('dcterms:title')[0].firstChild.wholeText
+
+    def get_dryad_title(self, doi='doi:10.5061/dryad.1850'):
+        metadata = self.get_dryad_metadata(doi)
+        return self.extract_dryad_title(metadata)
+
+    def check_dryad_doi(self, doi='doi:10.5061/dryad.1850'):
+        """ Checks dryad dois for validity.
+        Attempts to download data from
+        :param doi: Dryad DOI in the form of "doi:10.5061/dryad.XXXX"
+        :type doi: string
+        :returns:  bool -- True if the doi is found in the dryad archive.
+        :raises: urllib.error.HTTPError
+        """
+        dryad_metadata_path = furl(DRYAD_BASE_URL)
```

Comment:
This check is failing for me when doing my local testing.  I think the base url for this check (and the one in `get_dryad_metadata`) changed to `http://api.datadryad.org/`